### PR TITLE
Replaced ls_npq_tsf_details with ls_npq_tsf_application_detail

### PR DESCRIPTION
**Pull request by AntPage1**
Replaced ls_npq_tsf_details with ls_npq_tsf_application_detail which includes the updated logic relating to TSF eligibility criteria. This table is to be used in the NPQTSF Looker dashboard as well as 2 future Marts.

### CODE REVIEW

**Review by consult1815 — CHANGES_REQUESTED** [review]
I'm not sure we need two marts here as most of the code is identical. Do you think it would be possible to amalgamate the marts into one, with all the necessary checks, just omitting the WHERE statements and then add instructions in the table description  (and flag in field descriptions as well) on how to get the results of one of the two marts and then in Looker Studio add the requisite filters to get the desired output?

File: definitions/marts/looker_studio_marts/npq/ls_npq_tsf_application_detail.sqlx [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    ,IF(add.tsf_primary_plus_eligibility = TRUE, TRUE, app.tsf_primary_plus_eligibility) AS tsf_primary_plus_eligibility
+    ,IF(add.targeted_delivery_funding_eligibility = TRUE, TRUE, app.targeted_delivery_funding_eligibility) AS targeted_delivery_funding_eligibility
+
+    ,CASE
+      WHEN ex.application_id IS NOT NULL THEN 1 ELSE 0
+    END AS is_excluded
+
+    ,CASE
+      WHEN ex.application_id IS NOT NULL THEN ex.exclusion_reason 
+      WHEN app.targeted_delivery_funding_eligibility = FALSE AND add.application_id IS NULL THEN ''
```
[the remark this answers is not in the document]
Comment (AntPage1) [reply definitions/marts/looker_studio_marts/npq/ls_npq_tsf_application_detail.sqlx:140]: In the output Nathan requested that only the manually provided reason appears, previously I had 'Logic Unsatisfied'. The blank is used to identify those where the logic is satisfied and NULL is for any uncategorised rows which would require investigation to see why.

File: definitions/marts/looker_studio_marts/npq/ls_npq_tsf_application_detail.sqlx [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+    ,CASE
+      WHEN add.application_id IS NOT NULL THEN 1
+      WHEN app.targeted_delivery_funding_eligibility = TRUE AND ex.application_id IS NULL THEN 1
+      ELSE 0
+    END AS is_included
+
+    ,CASE
+      WHEN add.application_id IS NOT NULL THEN add.inclusion_reason
+      WHEN app.targeted_delivery_funding_eligibility = TRUE AND ex.application_id IS NULL THEN ''
```
[the remark this answers is not in the document]
Comment (AntPage1) [reply definitions/marts/looker_studio_marts/npq/ls_npq_tsf_application_detail.sqlx:152]: In the output Nathan requested that only the manually provided reason appears, previously I had 'Logic Satisfied'. The blank is used to identify those where the logic is satisfied and NULL is for any uncategorised rows which would require investigation to see why.

File: definitions/marts/looker_studio_marts/npq/ls_npq_tsf_application_detail.sqlx [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+      ELSE NULL
+    END AS inclusion_reason
+
+    ,CASE
```
[the remark this answers is not in the document]
Comment (AntPage1) [reply definitions/marts/looker_studio_marts/npq/ls_npq_tsf_application_detail.sqlx:156]: Lines 144-148 create a column of 1s & 0s for is_included. Lines 150-154 create a column of the inclusion_reason. The conditions in both case statements are the same but the output differs. If there is a means of apply the logic for 2 outcomes without duplication I'm happy to change to that method.

File: definitions/marts/looker_studio_marts/npq/ls_npq_tsf_applications_to_review.sqlx [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+    trn_verified,
+    cohort,
+    application_ecf_id AS application_id,
+    course_identifier,
+    school_urn,
+    school_name,
+    number_of_pupils,
+    primary_establishment,
+    targeted_delivery_funding_eligibility,
+    tsf_primary_eligibility, -- Unused by the service as targeted_delivery_funding_eligibility also acts as the flag for this indicator. Included purely for visibility.
```
[the remark this answers is not in the document]
Comment (AntPage1) [reply definitions/marts/looker_studio_marts/npq/ls_npq_tsf_applications_to_review.sqlx:73]: Marts will be combined into 1 and this document will be removed.

File: definitions/marts/looker_studio_marts/npq/ls_npq_tsf_applications_to_review.sqlx [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+    targeted_delivery_funding_eligibility IS FALSE
+    AND 
+    started_declaration_state IN (
+      'eligible',
+      'payable',
+      'paid')
+    AND 
+    started_declaration_date >= '2023-10-01'
+    AND
+    course_identifier NOT IN ('npq-early-headship-coaching-offer', 'npq-additional-support-offer') --EHCO/ASO courses are excluded as per instruction on 16/05/2024.
```
[the remark this answers is not in the document]
Comment (AntPage1) [reply definitions/marts/looker_studio_marts/npq/ls_npq_tsf_applications_to_review.sqlx:87]: Marts will be combined into 1 and this document will be removed.

File: definitions/marts/looker_studio_marts/npq/ls_npq_tsf_application_detail.sqlx [hunk 6]
```diff
@@ -3,13 +3,18 @@ config {
     assertions: {
         uniqueKey: ["application_id"]
     },
```
Comment (consult1815) [inline definitions/marts/looker_studio_marts/npq/ls_npq_tsf_application_detail.sqlx:5]: I think we should cluster by cohort, you can see how to do that in the npq_enrolment code in dataform

**Review by consult1815 — CHANGES_REQUESTED** [review]

**Review by consult1815 — APPROVED** [review]
Looks good

**Pull request merged**